ql/src/java/org/apache/hadoop/hive/ql/plan/ConditionalResolverMergeFiles.java

```diff
@@ -229,11 +249,20 @@ public List<Task<?>> getTasks(HiveConf conf, Object objCtx) {
   private void generateActualTasks(HiveConf conf, List<Task<?>> resTsks,
       long trgtSize, long avgConditionSize, Task<?> mvTask,
       Task<?> mrTask, Task<?> mrAndMvTask, Path dirPath,
-      FileSystem inpFs, ConditionalResolverMergeFilesCtx ctx, MapWork work, int dpLbLevel)
+      FileSystem inpFs, ConditionalResolverMergeFilesCtx ctx, MapWork work, int dpLbLevel,
+      boolean manifestFilePresent)
       throws IOException {
     DynamicPartitionCtx dpCtx = ctx.getDPCtx();
-    // get list of dynamic partitions
-    List<FileStatus> statusList = HiveStatsUtils.getFileStatusRecurse(dirPath, dpLbLevel, inpFs);
+    List<FileStatus> statusList;
+    if (manifestFilePresent) {
+      // Get the list of files from manifest file.
+      List<FileStatus> fileStatuses = getManifestFilePaths(conf, dirPath);
+      // Setup the work to include all the files present in the manifest.
+      setupWorkWhenUsingManifestFile(work, fileStatuses, dirPath);
+      statusList = HiveStatsUtils.getFileStatusRecurse(Utilities.toTempPath(dirPath), dpLbLevel, inpFs);
```

Comment:
Why do we need to `HiveStatsUtils.getFileStatusRecurse` again?

Reply:
This is to fetch the list of subdirectories under the table directory.

Reply:
Removed this as well, and identified directories using the manifest file itself.